src/utils/makeReactNativeConfig.js

```diff
@@ -237,6 +248,59 @@ function makeReactNativeConfig(
   return config;
 }
 
+/**
+ * Creates an array of configs based on changing `env` for every
+ * platform and returns
+ */
+function makeReactNativeConfig(
+  userWebpackConfig: WebpackConfigFactory,
+  options: ConfigOptions,
+  platform: Platform
+) {
+  /**
+   * We should support also the old format of config
+   * 
+   * module.exports = {
+   *   entry: './index.js',
+   * };
+   * 
+   * module.exports = ({ platform }) => ({
+   *   entry: `./index.${platform}.js`,
+   * });
+   */
+  if (!Object.prototype.hasOwnProperty.call(userWebpackConfig, 'webpack')) {
```

Comment:
Anyone has better idea how to check "old format" of the config file? @satya164 @zamotany

Reply:
@jukben it'll be webpack.haul.js whereas new one is haul.config.js

Reply:
That's only defaulted config, we support passing whatever file name user like. So check by a name file is definitely not a bullet-proof idea.

Reply:
What do you mean by "we support passing whatever file name"? How can you specify a file with a different name for the config? I am not aware of a way, and the docs don't mention of any https://github.com/callstack/haul/blob/master/docs/Configuration.md

Reply:
Yeah, you are able to `yarn haul start --config webpack.haul.js`

Reply:
https://github.com/callstack/haul/blob/master/src/commands/start.js#L189 the same for bundle - and I like this feature, we should document it

Reply:
Interesting. Ok so seems `const isLegacy = typeof config === 'function' || !config.webpack;` should be enough.

Reply:
Why not to stick with `Object.prototype.hasOwnProperty.call(userWebpackConfig, 'webpack')`?